packages/indicators/src/builders/buildArithmetic.ts

```diff
@@ -3,44 +3,84 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { getVariables, runArithmetic } from '@beyondessential/arithmetic';
+
 import { Aggregator } from '@tupaia/aggregator';
-import {
-  getAggregationsByCode,
-  extractDataElementCodesFromFormula,
-  fetchAnalytics,
-} from './helpers';
-import { Builder, AggregationSpecs, FetchOptions } from '../types';
-
-interface ArithmeticConfig {
+import { hasContent, isAString, isPlainObject } from '@tupaia/utils';
+import { getAggregationsByCode, fetchAnalytics, validateConfig } from './helpers';
+import { Analytic, Builder, AggregationSpecs, FetchOptions } from '../types';
+
+export interface ArithmeticConfig {
+  readonly [key: string]: unknown;
```

Comment:
You are correct, we shouldn't expect other fields here (although we don't exclude them during validation...) This change was a result of adhering to type definitions in `buildArithmetic.test.ts`.

Long story short: `const buildArithmetic: Builder` expects an `Record<string, unknown>` config according to our type definitions. The `ArithmeticConfig` **interface** cannot be passed as `Record<string, unknown>`, since interfaces can be extended and new keys of `number` type could exist in a variable implementing `ArithmeticConfig`. This would break the `string` key expectation in `Record<string, unknown>`.

What I did here solves this issue but indeed opens up the interface to any `string` key. It seems that another solution is defining `ArithmeticConfig` as a **type** instead, which locks in the supported keys, ensuring that their type will be `string`.